framework/auth/core.py

```diff
@@ -658,6 +660,8 @@ def set_password(self, raw_password, notify=True):
         """
         had_existing_password = bool(self.password)
         self.password = generate_password_hash(raw_password)
+        if self.username == raw_password:
+            raise HTTPError(httplib.CONFLICT)
```

Comment:
Please include a descriptive error message, e.g. `HTTPError(httplib.CONFLICT, message='You cannot set your password to your email')`